src/libmeasurement_kit/dns/system_resolver.cpp

```diff
@@ -1,18 +0,0 @@
-// Part of measurement-kit <https://measurement-kit.github.io/>.
-// Measurement-kit is free software. See AUTHORS and LICENSE for more
-// information on the copying conditions.
-
-#include "../dns/system_resolver_impl.hpp"
-
-namespace mk {
-namespace dns {
-
-void system_resolver(QueryClass dns_class, QueryType dns_type, std::string name,
-                     Callback<Error, Var<Message>> cb, Settings settings,
-                     Var<Reactor> reactor, Var<Logger> logger) {
-    system_resolver_impl(dns_class, dns_type, name, cb, settings, reactor,
-            logger);
-}
-
-} // namespace dns
-} // namespace mk
```

Comment:
Code removed. @AntonioLangiu initially organized the code to have everything into a single private header. Later, I started splitting code in foo.hpp, foo_impl.hpp, and foo.cpp. The argument for doing that was that you catch compile errors early because you instantiate templates in the related filed. But, after reading again the code, I realized I like more the approach of @AntonioLangiu, because it's more compact. So, I'm basically moving again everything into a header.